### PR TITLE
No parent input if .spawn

### PR DESCRIPTION
**Pull request by gongy**

### CODE REVIEW

File: modal/functions.py [hunk 1]
```diff
@@ -235,13 +235,14 @@ def __init__(self, stub: api_grpc.ModalClientStub, function_call_id: str, client
         self.function_call_id = function_call_id  # TODO: remove and use only input_id
 
     @staticmethod
-    async def create(function_id: str, args, kwargs, client: _Client) -> "_Invocation":
+    async def create(function_id: str, detached: bool, args, kwargs, client: _Client) -> "_Invocation":
         assert client.stub
         item = await _create_input(args, kwargs, client)
 
+        parent_input_id = current_input_id() if detached else None
```
Comment (aksh-at) [inline modal/functions.py:242]: Should this be the other way around?

File: modal/functions.py [hunk 2]
```diff
@@ -235,13 +235,14 @@ def __init__(self, stub: api_grpc.ModalClientStub, function_call_id: str, client
         self.function_call_id = function_call_id  # TODO: remove and use only input_id
 
     @staticmethod
-    async def create(function_id: str, args, kwargs, client: _Client) -> "_Invocation":
+    async def create(function_id: str, nowait: bool, args, kwargs, client: _Client) -> "_Invocation":
```
Comment (freider) [inline modal/functions.py:238]: nit: `nowait` isn't immediately obvious what effects it has - maybe `set_parent` or something? Is this related to whether we should kill child calls when the parent is terminated or not?

Comment (mwaskom) [reply modal/functions.py:238]: +1

**Comment by mwaskom** [comment]
Closing as stale and about to be conflicted by @freider's container pool refactor. Feel free to reopen if still relevant.

**Pull request closed without merging**